classify.py

```diff
@@ -313,58 +335,121 @@ def evaluate_samesource(desc, dataset, n_frequent_words, tokens_per_sample, max_
     LOG.info(f'{desc}: number of speakers: {np.unique(y).size}')
     LOG.info(f'{desc}: number of instances: {y.size}')
 
-    X = preprocessor.fit_transform(X)
+    X = preprocessor.fit_transform(X) # shouldn't this take place on the x_train and then on x_test?
 
     lrs = []
+    lrs_voc = []
+    lrs_comb = []
+    lrs_features = []
     y_all = []
-    for X_train, y_train, X_test, y_test in tqdm(get_batch_simple(X, y, repeats, max_n_of_pairs_per_class)):
-        # fit a classifier and calculate LRs
+
+    for X_train, y_train, X_test, y_test, X_voc_train, X_voc_test in tqdm(
+            get_batch_simple(X, y, X_voc, conv_ids, repeats,
+                             max_n_of_pairs_per_class)):
+
+        # there are three ways to combine the mfw method with the voc output
+        # 1. assume that mfw and voc LR scores are independent and multiply them (m1)
+        # 2. apply logit using as input the mfc and the voc score, then calibrate the resulted score (m2)
+        # 3. use the voc score as additional feature to the mfw input vector (m3)
+        # NOTE: m3 can be used only if the scorer is a classification alg, otherwise m3 = m2 with scorer always a logit
+
+        # calculate LRs for vocalize output (for m1)
+        voc_cal.fit(X=X_voc_train, y=y_train)
+        lrs_voc.append(voc_cal.transform(X_voc_test))
+
+        # mfw - fit a classifier and calculate LRs (for m1, the resulted scorer can also be used for m2)
         clf.fit(X_train, y_train)
         lrs.append(clf.predict_lr(X_test))
         y_all.append(y_test)
 
+        # take scores from mfw scorer and combine with voc output using logit then calibrate (for m2)
+        mfw_scores_train = lir.apply_scorer(clf.scorer, X_train)
+        X_comb_train = np.vstack((np.squeeze(mfw_scores_train), X_voc_train)).T
+        mfw_voc_clf.fit(X_comb_train, y_train)
```

Comment:
`mfw_scores_train` is a score in range 0..1
`X_voc_train` is an LR, log LR or something else?

Classifier may work better if both scores are in the same range, for example 0..1, or -1..1 (may require scaling)

Reply:
done